### PR TITLE
Fix MmapShMemProvider's name

### PR DESCRIPTION
**Pull request by tokatoka**
rename MmapShMem's name from "/libafl-{proc_id}-{ctr}" to "/libafl-{proc_id}-{random u32}"

`"/libafl_{proc_id}_{ctr}"` is a problem
If we create 2 MmapShMem in the same process, and both called MmapShMem::new()
They have the same process id and the same counter. I changed it to a random number so they don't collide

### CODE REVIEW

File: libafl/src/bolts/shmem.rs [hunk 1]
```diff
@@ -53,7 +53,7 @@ pub type StdShMemService = ShMemService<MmapShMemProvider>;
     unix,
     not(any(target_os = "android", target_vendor = "apple"))
 ))]
-pub type StdShMemProvider = UnixShMemProvider;
+pub type StdShMemProvider = MmapShMemProvider;
```
Comment (tokatoka) [inline libafl/src/bolts/shmem.rs:56]: For test

File: libafl/src/bolts/shmem.rs [hunk 2]
```diff
@@ -588,6 +593,10 @@ pub mod unix_shmem {
             pub __glibc_reserved5: c_ulong,
         }
 
+        #[cfg(target_vendor = "apple")]
+        const MAX_MMAP_FILENAME_LEN: usize = 32;
```
Comment (tokatoka) [inline libafl/src/bolts/shmem.rs:597]: This is macOS's limit

File: libafl/src/bolts/shmem.rs [hunk 3]
```diff
@@ -84,6 +84,10 @@ impl ShMemDescription {
 
 /// An id associated with a given shared memory mapping ([`ShMem`]), which can be used to
 /// establish shared-mappings between proccesses.
+/// Id is a file descriptor if you use `MmapShMem` or `AshmemShMem`.
```
Comment (domenukk) [inline libafl/src/bolts/shmem.rs:87]: Why a file descriptor for MmapShMem? I thought it's a file on disk?

File: libafl/src/bolts/shmem.rs (hunk 2 above)
Comment (domenukk) [reply libafl/src/bolts/shmem.rs:597]: Maybe add this information as comment? Also, it's 31 chars according to the link you posted, not 32?

Comment (tokatoka) [reply libafl/src/bolts/shmem.rs:597]: fixed

File: libafl/src/bolts/shmem.rs (hunk 3 above)
Comment (tokatoka) [reply libafl/src/bolts/shmem.rs:87]: Yeah, but what we are actually storing into ShMemId.Id is the file descriptor atm. (else unix_shmem_server does not work)
The underlying file's name is not put into `Id` itself

**Pull request merged**